xmpp/auth.py

```diff
@@ -19,28 +19,27 @@
 Can be used both for client and transport authentication.
 """
 
-import sys
 from .protocol import *
 from .client import PlugIn
 import base64,random,re
 from . import dispatcher
 from hashlib import md5,sha1
+from six import ensure_str,ensure_binary
 
-IS_PYTHON3 = sys.version_info >= (3, 0, 0)
+CHARSET_ENCODING='utf-8'
 
 def HH(some):
-    if IS_PYTHON3 and isinstance(some, str):
-        some = some.encode("utf-8")
-    return md5(some).hexdigest()
+    return md5(ensure_binary(some, CHARSET_ENCODING)).hexdigest()
 def H(some):
-    if IS_PYTHON3 and isinstance(some, str):
-        some = some.encode("utf-8")
-    return md5(some).digest()
+    return md5(ensure_binary(some, CHARSET_ENCODING)).digest()
 def C(some):
-    if IS_PYTHON3:
-        some = [x.encode('utf-8') if isinstance(x, str) else x for x in some]
-        return b':'.join(some)
-    return ':'.join(map(str, some))
+    some = [ensure_binary(x, CHARSET_ENCODING) for x in some]
+    return b':'.join(some)
+
+def HHSHA1(some):
+    return sha1(ensure_binary(some, CHARSET_ENCODING)).hexdigest()
+def B64(some):
+    return ensure_str(base64.b64encode(ensure_binary(some,CHARSET_ENCODING)),CHARSET_ENCODING).replace('\r', '').replace('\n', '')
```

Comment:
Removing newlines with `replace` shouldn't be required after the switch from `encodestring` to `b64encode` in #37 and can be removed.

Reply:
Thanks. I had the same thought, but it's nice that you confirmed it. I've amended the patch accordingly.